readiness-tool/popup.js

```diff
@@ -0,0 +1,231 @@
+self.popups = {};
+self.popups.isSupported = isSupported;
+self.popups.doesRegexExist = doesRegexExist;
+
+/**
+ * Callback function that sends a response upon receiving message
+ * @param {!Object} request - Message Object
+ * @param {!Object} sender - Message sender defined here
+ */
+chrome.runtime.onMessage.addListener(function (request, sender) {
+  if (request.action == "getSource") {
+    // htmlOfTab contains the page source as a string
+    htmlOfTab = request.source;
+    findDetectedApps(htmlOfTab);
+  }
+});
+
+/** Event called when window is loaded */
+function onWindowLoad() {
+  var htmlOfTab;
+  // When page is loaded, display "Loading..." so the user expects content
+  const supportedAds = document.getElementById("ads-supported");
+  const supportedAnalytics = document.getElementById("analytics-supported");
+  const notSupportedAds = document.getElementById("ads-notSupported");
+  const notSupportedAnalytics = document.getElementById("analytics-notSupported");
+  const notSupportedOther = document.getElementById("other-notSupported");
+  supportedAds.innerHTML = 'Loading...';
+  supportedAnalytics.innerHTML = "Loading...";
+  notSupportedAds.innerHTML = "Loading...";
+  notSupportedAnalytics.innerHTML = "Loading...";
+  notSupportedOther.innerHTML = "Loading...";
+
+  // Gets the DOM of the current web page and converts it to a string
+  chrome.tabs.executeScript(null, {
+    file: "getPagesSource.js"
```

Comment:
From my understanding of the chrome extension, I have to get the source code of the active Tab from the "background script". Otherwise when I do something like document.documentElement.outerHTML I'd be getting the HTML of the chrome extension.